components/datetime/src/fields/symbols.rs

```diff
@@ -235,7 +235,8 @@ impl TryFrom<char> for FieldSymbol {
         .or_else(|_| Day::try_from(ch).map(Self::Day))
         .or_else(|_| Weekday::try_from(ch).map(Self::Weekday))
         .or_else(|_| DayPeriod::try_from(ch).map(Self::DayPeriod))
-        .or_else(|_| Hour::try_from(ch).map(Self::Hour)).or({
+        .or_else(|_| Hour::try_from(ch).map(Self::Hour))
+        .or({
```

Comment:
Thought: it's weird that `fmt` doesn't remove the extraneous block. Also, why is clippy replacing `or_else`s with `or`s now?

Reply:
it only does it when it knows that the thing in the closure has zero cost to construct